pystac_client/mixins.py

```diff
@@ -32,14 +32,24 @@ def _get_href(self, rel: str, link: Optional[pystac.Link], endpoint: str) -> str
 class QueryablesMixin(BaseMixin):
     """Mixin for adding support for /queryables endpoint"""
 
-    def get_queryables(self) -> Dict[str, Any]:
-        """Return all queryables.
+    def get_queryables(self, *collections: Optional[str]) -> Dict[str, Any]:
+        """Return all queryables, or limit to those of specified collections.
 
         Output is a dictionary that can be used in ``jsonshema.validate``
+        Args:
+            *collections: The IDs of the items to include.
 
         Return:
             Dict[str, Any]: Dictionary containing queryable fields
         """
+        if collections and isinstance(self, pystac.Catalog):
+            response = self.get_collection(collections[0]).get_queryables()
```

Comment:
I think this does a double-request, i.e.:

```text
GET /collections/foo
GET /collections/foo/queryables
```

This is one of my problems with [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS) patterns sometimes -- sometimes you know what you want, and can do less work to go get it.

Pinging @jsignell, who did the excellent [redesign of href resolution](https://github.com/stac-utils/pystac-client/pull/480), for thoughts. Core question: **is there any way, in the new href resolution regime, to get the queryables href without fetching the collection first?** I want to avoid scattering href building code all over the repo, like we had before, but two requests when you could just do one seems bad.

Reply:
My thought is that the `collections[0]` here is an unsubstantiated claim to the existence of a collection named `collections[0]`.  My thought was to check that it was a real collection, BUT having just given a shot in the browser, seems like skipping that check works with an appropriate:
```
{
  "code": "NotFoundError",
  "description": "Collection banana not found"
}
```

Reply:
> My thought was to check that it was a real collection, BUT having just given a shot in the browser, seems like skipping that check works with an appropriate:

That _could_ be a implementation-specific response (saying that there's a missing collection when you hit that collection's queryable endpoint) but I think that's ok. It'd be _extremely_ surprising, and probably non-compliant, to 200 on `/collections/foo/queryables` but 404 on `/collections/foo`.